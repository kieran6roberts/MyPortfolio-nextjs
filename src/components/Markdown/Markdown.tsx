import * as React from "react";
import ReactMarkdownWithHtml from "react-markdown/with-html";

export default function Markdown({ string }: { string: string}) {
    return (
        <article className="text-xs">
            <ReactMarkdownWithHtml children={string} allowDangerousHtml />
        </article>
    )
};