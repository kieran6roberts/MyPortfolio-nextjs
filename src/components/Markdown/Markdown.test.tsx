import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import Markdown from "@/components/Markdown/Markdown";

let documentBody: RenderResult;

describe("<Markdown /> converts markdown to html", () => {
    const markdown = `## Markdown`;
    
    test("renders", () => {
        documentBody = render(<Markdown string={markdown} />);

        expect(documentBody.getByRole("heading", { name: /markdown/i })).toBeInTheDocument();
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<Markdown string={markdown} />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});