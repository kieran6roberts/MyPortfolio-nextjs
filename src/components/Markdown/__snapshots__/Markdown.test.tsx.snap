// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> converts markdown to html render snapshot 1`] = `
<article
  className="text-xs"
>
  <h2>
    Markdown
  </h2>
</article>
`;
